crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
I was thinking of doing `CRYPTO_set_ex_data(to, mx, NULL)` to avoid the overwrite issue, but that would likely increase the array size, which is not necessarily a bad thing.